django_fsm_freeze/models.py

```diff
@@ -103,12 +127,15 @@ def _get_fsm_field(cls) -> models.Field:
     @classmethod
     def config_check(cls) -> None:
         errors = defaultdict(list)
-        try:
-            cls._get_fsm_field()
-        except FieldDoesNotExist:
-            errors['FROZEN_STATE_LOOKUP_FIELD'].append('FSMField not found.')
-        except TypeError as err:
-            errors['FROZEN_STATE_LOOKUP_FIELD'].append(str(err))
+        if not cls.FROZEN_DELEGATE_TO:
```

Comment:
if `cls.FROZEN_DELEGATE_TO` is given, do we want to validate the value here?

Reply:
If we do, propose also adding a test for that a `FreezeConfigurationError` is raised properly.

```
 def test_frozen_delegation_to_not_a_valid_value(self, active_fake_obj):
        sub_fake = SubFakeModel.objects.create(fake_model=active_fake_obj)
        sub_sub_fake = SubSubFakeModel.objects.create(sub_fake_model=sub_fake)

        setattr(SubSubFakeModel, 'FROZEN_DELEGATE_TO', 'a.nonsense.field')

        with pytest.raises(FreezeConfigurationError) as err:
            sub_sub_fake.config_check()

        assert err......

```

Reply:
What if both `FROZEN_STATE_LOOKUP_FIELD` and `FROZEN_DELEGATE_TO` are given in the same class, e.g. 
```
class SubSubFakeModel(FreezableFSMModelMixin):
    FROZEN_DELEGATE_TO = 'sub_fake_model.fake_model'  # <== here
    FROZEN_STATE_LOOKUP_FIELD = 'status'               #  <== and here
    
    NON_FROZEN_FIELDS = ('can_change_me',)

    status = FSMField(default=FakeStates.NEW.value)    #  <== assuming SubSubFake also has its own FSMField
 
    sub_fake_model = models.ForeignKey(SubFakeModel, on_delete=models.PROTECT)
    can_change_me = models.BooleanField(default=False)
    cannot_change_me = models.BooleanField(default=False)
```

Reply:
> if `cls.FROZEN_DELEGATE_TO` is given, do we want to validate the value here?

I can try.

Reply:
> What if both `FROZEN_STATE_LOOKUP_FIELD` and `FROZEN_DELEGATE_TO` are given in the same class, e.g.
> 
> ```
> class SubSubFakeModel(FreezableFSMModelMixin):
>     FROZEN_DELEGATE_TO = 'sub_fake_model.fake_model'  # <== here
>     FROZEN_STATE_LOOKUP_FIELD = 'status'               #  <== and here
>     
>     NON_FROZEN_FIELDS = ('can_change_me',)
> 
>     status = FSMField(default=FakeStates.NEW.value)    #  <== assuming SubSubFake also has its own FSMField
>  
>     sub_fake_model = models.ForeignKey(SubFakeModel, on_delete=models.PROTECT)
>     can_change_me = models.BooleanField(default=False)
>     cannot_change_me = models.BooleanField(default=False)
> ```

They should be exclusive. it's like having `FROZEN_DELEGATE_TO` and `NON_FROZEN_FIELDS` defined together. There is no use case for it.

Reply:
Yeah, I think so too. So it'd be better to have a config check on that I think.

Reply:
we cannot fully valildate the value of FROZEN_DELEGATE_TO, because the models are not fully loaded yet, at the time of execution.

```
raise AppRegistryNotReady("Models aren't loaded yet.")
```
I suggest we let the checks fail at runtime instead of load time

Reply:
> we cannot fully validate the value of FROZEN_DELEGATE_TO ...
> I suggest we let the checks fail at runtime instead of load time

I see, for the value yes. I agree then let's do it like this.

How about the case where both `FROZEN_STATE_LOOKUP_FIELD` and `FROZEN_DELEGATE_TO` are both being defined? Just make sure at most only one of them is set (not checking the value itself)?

Reply:
yeah it's done.
https://github.com/ming-tung/django-fsm-freeze/pull/7/files#diff-ffcc37ec84a405276b01f8add40e69b44fd43fd627074e906f041b06db49fb94R158